### PR TITLE
Plan9

### PR DESCRIPTION
**Pull request by rminnich**
This is being used for Harvey.

### CODE REVIEW

**Comment by rminnich** [comment]
marked code health as I worry a lot about linux-isms creeping in.

File: pkg/memio/ports_linux.go [hunk 1]
```diff
@@ -16,17 +16,20 @@ var (
 	linuxPath = "/dev/port"
 )
 
+// LinuxPort implements MemIOReadWriteCloser for Linux devices.
```
Comment (MDr164) [inline pkg/memio/ports_linux.go:19]: Shouldn't that be `implements ReadWriterCloser` as it got renamed?

File: cmds/core/cp/cp_others.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !plan9
+// +build !plan9
```
Comment (MDr164) [inline cmds/core/cp/cp_others.go:6]: I think the `+build` line is not necessary as this is legacy behavior for the Go version we're targeting

File: pkg/memio/memio_ports.go [hunk 3]
```diff
@@ -12,45 +12,52 @@ import (
...
-type MemIOPort struct {
+// Port implements memory and IO port access via an os.File.
+type Port struct {
 	*os.File
 }
 
-func (m *MemIOPort) Read(out UintN, addr int64) error {
+var _ ReadWriteCloser = &Port{}
+
+func (m *Port) Read(out UintN, addr int64) error {
```
Comment (MDr164) [inline pkg/memio/memio_ports.go:39]: Maybe add a comment above this function as well, now that you're adding them to other functions in this file.

File: pkg/memio/memio_ports.go [hunk 4]
```diff
@@ -12,45 +12,52 @@ import (
...
 	if _, err := m.File.Seek(addr, io.SeekStart); err != nil {
 		return err
 	}
 	return binary.Read(m.File, ubinary.NativeEndian, out)
 }
 
-func (m *MemIOPort) Write(in UintN, addr int64) error {
+func (m *Port) Write(in UintN, addr int64) error {
```
Comment (MDr164) [inline pkg/memio/memio_ports.go:46]: see above

File: cmds/core/cp/cp_others.go (hunk 2 above)
Comment (rminnich) [reply cmds/core/cp/cp_others.go:6]: gofmt puts it in automagically, so I think I need to leave it there.

**Pull request merged**